float32 percentageLeft
float32 maxVoltage